sdk/typescript/src/verify/index.ts

```diff
@@ -53,26 +57,38 @@ export async function verifyTransactionBlock(
 	return parsedSignature.publicKey;
 }
 
-function parseSignature(signature: SerializedSignature) {
+export function parseSignature(signature: SerializedSignature) {
 	const parsedSignature = parseSerializedSignature(signature);
-	let publicKey: PublicKey;
 
-	switch (parsedSignature.signatureScheme) {
-		case 'ED25519':
-			publicKey = new Ed25519PublicKey(parsedSignature.publicKey);
-			break;
-		case 'Secp256k1':
-			publicKey = new Secp256k1PublicKey(parsedSignature.publicKey);
-			break;
-		case 'Secp256r1':
-			publicKey = new Secp256r1PublicKey(parsedSignature.publicKey);
-			break;
-		default:
-			throw new Error(`Unsupported signature scheme ${parsedSignature.signatureScheme}`);
+	if (parsedSignature.signatureScheme === 'MultiSig') {
+		const signatureBytes = parsedSignature.bytes.slice(1);
+		const multisig: MultiSigStruct = builder.de('MultiSig', signatureBytes);
+		return {
+			...parsedSignature,
+			multisig,
+			publicKey: new MultiSigPublicKey(multisig.multisig_pk),
+			signature: signatureBytes,
```

Comment:
this is tied to how verify on a multisig publickey works, there is no single signature to extract from the serialized signature, so we are using the bytes of the full serializedSignature excluding the flag here.


This may be a bad idea, but I was considering if it would make sense to have all PublicKey classes know how to verify serializedSignatures, so the interface would be something like `verify(data: UInt8Array, signature: SerializedSignature | UInt8Array) => Promise<boolean>` and then have the MultiSigPublicKey throw if you try to verify with bytes rather than the serialized signature.